src/libDirectoryService/DSBlockPostProcessing.cpp

```diff
@@ -359,6 +359,11 @@ void DirectoryService::StartFirstTxEpoch() {
   Blacklist::GetInstance().Pop(BLACKLIST_NUM_TO_POP);
   P2PComm::ClearPeerConnectionCount();
 
+  {
+    lock_guard<mutex> g(m_mediator.m_node->m_mutexWhitelistReqs);
```

Comment:
For easier maintenance (and better encapsulation) might be better to have a function to both acquire the mutex and clear the map